ui/lib/core/addon/utils/search-select-has-many.js

```diff
@@ -15,19 +15,23 @@
  * @param modelRecord string passed to peekRecord
  */
 
-export default function handleHasManySelection(selectedIds, modelCollection, store, modelRecord) {
+export default function handleHasManySelection(selectedIds, modelCollection, store, modelRecord, model) {
   // first check for existing models that have been removed from selection
   modelCollection.forEach((model) => {
-    if (!selectedIds.includes(model.id)) {
+    const modelAttr = model?.id || model?.name;
+    if (!selectedIds.includes(modelAttr)) {
       modelCollection.removeObject(model);
     }
   });
   // now check for selected items that don't exist and add them to the model
   const modelIds = modelCollection.mapBy('id');
   selectedIds.forEach((id) => {
     if (!modelIds.includes(id)) {
-      const model = store.peekRecord(modelRecord, id);
+      let model = store.peekRecord(modelRecord, id);
```

Comment:
Since we're passing in `model` as a param, I would change either the param name or this and line 34. This line is actually safe as a let within the `if` but line 34 is returning to the function param `model` so this may not behave as expected. 
Also, let's make sure we have good test coverage for this util

Reply:
yes - agreed on test coverage! 

i was having issues with the model saving, but through _lots_ of trial, I think I fixed the issue and was able to remove model as a param (which I didn't totally love doing in the first place).